sys/net/gnrc/netif/ieee802154/gnrc_netif_ieee802154.c

```diff
@@ -436,4 +437,46 @@ static int _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)
     }
     return res;
 }
+
+int _get(gnrc_netif_t *netif, gnrc_netapi_opt_t *opt)
+{
+    int res = -ENOTSUP;
+
+    gnrc_netif_acquire(netif);
+    switch (opt->opt) {
+        case NETOPT_MAX_SDU_SIZE:
+            if (opt->data_len >= sizeof(iolist_t)) {
+                const iolist_t *in = opt->data;
+                const gnrc_netif_hdr_t *l2 = in->iol_base;
+                gnrc_netapi_opt_t opt_dev = { .context = 0 };
+                uint16_t sdu;
+                opt_dev.data = &sdu;
+                opt_dev.data_len = sizeof(sdu);
+                opt_dev.opt = NETOPT_MAX_PDU_SIZE;
+                if (gnrc_netif_get_from_netdev(netif, &opt_dev) <= 0) {
+                    goto release;
+                }
+                uint8_t mhr[IEEE802154_MAX_HDR_LEN];
+                int hdr_len = gnrc_netif_hdr_to_ieee802154(mhr, netif, l2);
+                if (hdr_len < 0) {
+                    goto release;
+                }
+                sdu += (IEEE802154_MAX_HDR_LEN - hdr_len);
```

Comment:
better add a comment here that `NETOPT_MAX_PDU_SIZE` returns the most pessimistic PDU size and here the calculation is reversed